### PR TITLE
Move to using JSON file in new deployments repo

### PR DESCRIPTION
**Pull request by Tritium-VLK**
This used to pull data from another repo I wrote to crawl the deployments directory structure.

Now @jubeira has created a version of this file in the deployments repo that is directly maintained by blabs.

This PR changes the contracts autogen to use this file directly.  

It also includes a run, to demonstrate only real changes were applied.

Right now it's a PR into the #86.  Would maybe make more sense to merge this into that, and then the whole #86.  Lots lingering and things are getting out of data.  Think we could review and merge soon?

@jubeira: Mind taking a look at the changes to the logic and giving it a quick review?

### CODE REVIEW

File: .github/python_actions/genContractLists.py [hunk 1]
```diff
@@ -18,28 +19,36 @@
...
-   return r.json()
+def address_directory(chain, status=None):
+    r = requests.get(f"https://raw.githubusercontent.com/balancer/balancer-deployments/master/addresses/{chain}.json")
+    r=r.json()
+    if isinstance(status, str):
+        filtered = {}
+        for deployment, depdata in r.items():
+            if depdata["status"] == status:
+                filtered[deployment] = depdata
+        return filtered
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
        return {k: v for k, v in r.items() if v['status'] == status}
```

You can use dict comprehension or `filter` for a one-liner.

File: .github/python_actions/genContractLists.py [hunk 2]
```diff
@@ -18,28 +19,36 @@
...
+
 
 
 def genFullTable(r, chain):
     result = pd.DataFrame(columns=["Contract", "Address", "Deployment"])
-    r = r[chain]  # go to chain
-    for deployment, contracts in r.items():
-        for contract in contracts.keys():
+    for deployment, depdata in r.items():
+        for contract in depdata["contracts"]:
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
        for contract in depdata['contracts']:
```

File: .github/python_actions/genContractLists.py [hunk 3]
```diff
@@ -49,49 +58,47 @@ def genFullTable(r, chain):
...
     result = pd.DataFrame(columns=["Contract", "Address", "Deployment"])
     print(f"Generating pools for {chain}")
-    r = r[chain] # go to chain
-    for deployment, contracts in r.items():
+    for deployment, depdata in r.items():
         if "-pool" not in deployment:
             continue
-        for contract, address in contracts.items():
-            if "Factory" in contract:
+        for contract in depdata["contracts"]:
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
        for contract in depdata['contracts']:
```

File: .github/python_actions/genContractLists.py [hunk 4]
```diff
@@ -49,49 +58,47 @@ def genFullTable(r, chain):
...
     return result
 
 def genNotInContractList(r, chain, contractList):
     result = pd.DataFrame(columns=["Contract", "Address", "Deployment"])
-    r = r[chain] # go to chain
-    for deployment, contracts in r.items():
-        for contract, address in contracts.items():
-            if contract in contractList:
+    for deployment, depdata in r.items():
+        for contract in depdata["contracts"]:
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
        for contract in depdata['contracts']:
```

File: .github/python_actions/genContractLists.py [hunk 5]
```diff
@@ -49,49 +58,47 @@ def genFullTable(r, chain):
...
+            if contract['name'] in contractList:
                 continue
-            if "-pool"  in deployment:
+            if "-pool" in deployment:
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
            if '-pool' in deployment:
```

File: .github/python_actions/genContractLists.py [hunk 6]
```diff
@@ -101,24 +108,26 @@ def genNotInContractList(r, chain, contractList):
 
 def genFromContractList(r, chain, contractList):
     result = pd.DataFrame(columns=["Contract", "Address", "Deployment"])
-    r = r[chain] # go to chain
-    for deployment, contracts in r.items():
-        m = set(contracts.keys()) & set(contractList)
-        for contract in m:
+    for deployment, depdata in r.items():
+        for contract in depdata["contracts"]:
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
        for contract in depdata['contracts']:
```

File: .github/python_actions/genContractLists.py [hunk 7]
```diff
@@ -101,24 +108,26 @@ def genNotInContractList(r, chain, contractList):
...
+            ### Check if in list
+            if contract["name"] not in contractList:
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
            if contract['name'] not in contractList:
```

File: .github/python_actions/genContractLists.py [hunk 8]
```diff
@@ -137,7 +146,7 @@ def genChainMd(chain):
 ## Pool Factories
 
 """
-    r = address_directory()["active"]
+    r = address_directory(chain, status="ACTIVE")
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
    r = address_directory(chain, status='ACTIVE')
```

File: .github/python_actions/genContractLists.py [hunk 9]
```diff
@@ -162,7 +171,7 @@ def genChainMd(chain):
     
 """
     try:
-        r = address_directory()["old"]
+        r = address_directory(chain, status="DEPRECATED")
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]:
```suggestion
        r = address_directory(chain, status='DEPRECATED')
```

**Review by jubeira — APPROVED** [review]
Just a few style comments, otherwise LGTM!

**Comment by Tritium-VLK** [comment]
Thanks a lot @jubeira.  Accepted all your suggestions.  The "'s were def a bit messy, I should have cleaned that up.

There's not great(any) unit tests, but reran and no changes to the resulting output files, which is to be expected.

File: .github/python_actions/genContractLists.py [hunk 10]
```diff
@@ -162,7 +167,7 @@ def genChainMd(chain):
     
 """
     try:
-        r = address_directory()["old"]
+        r = address_directory(chain, status='DEPRECATED')
         output += genFullTable(r, chain).to_markdown(index=False)
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]: (see comment below regarding empty results from `address_directory`.

File: docs/reference/contracts/deployment-addresses/zkevm.md [hunk 11]
```diff
@@ -70,8 +69,8 @@ These deployments were in use at some point, and may still be in active operatio
 #### If you can only find the contract you are looking for in the deprecated section and it is not an old pool, try checking the deployments tasks to find it or ask in the Discord before using a deprecated contract.
 
     
-No deprecated contracts found
-
+| Contract   | Address   | Deployment   |
+|------------|-----------|--------------|
```
Comment (jubeira) [inline docs/reference/contracts/deployment-addresses/zkevm.md]: Ah, it looks like if you don't find anything the code doesn't throw an exception now, so you get an empty table.
If you want to print `No deprecated contracts found` as before maybe that case can be handled differently by checking whether the returned `dict` is empty or not.

File: .github/python_actions/genContractLists.py [hunk 12]
```diff
@@ -2,6 +2,7 @@
 import pandas as pd
 import re
 from bal_addresses import AddrBook
+from dotmap import DotMap
```
Comment (jubeira) [inline .github/python_actions/genContractLists.py]: Is this used anywhere? I don't see it referenced in the new changes.

**Review by jubeira — APPROVED** [review]
LGTM! Only a few super minor comments from the second pass for your consideration :)

**Comment by Tritium-VLK** [comment]
@mikemcdonald this is tested and working and needed now that things have moved from the monorepo.

Released CSPv5 today.  Ran a manual run from this branch to create #118 with all the backlogged updates.

The permissions thing is broken because of keys, but working to move that to bal_addresses so that the action in this repo doesn't need web3 keys and is simpler/faster.

For now it should work if you add the missing secret, but it can wait.

**Review by mikemcdonald — APPROVED** [review]

**Pull request merged**